### ISSUE TITLE
Generic gen_seed method

### ISSUE THREAD

**Issue by GregPlowman** [sunoru/RandomNumbers.jl#36]
I see that `gen_seed` is a common function for all RNGs in `RandomNumbers`. It would be useful to able to call `gen_seed(rng)` and have it return the appropriately sized tuple for that rng. This could make code more generic.
I can't see whether this is currently possible. My workaround has been:

```
Base.length(::Type{NTuple{N,T}}) where {N,T} = N
seed_length(rng) = length(seed_type(rng))
gen_seed(rng::RandomNumbers.AbstractRNG) = gen_seed(output_type(rng), seed_length(rng))
```

Not sure if `Base.length` is the right function to extend here, or if there already exists a function for this.

**Comment by GregPlowman** [comment]
> However I also believe it might be helpful to generate seeds manually so that the seeds could be stored for later use.

Yes, exactly. That is my use case. I want randomly-seeded but reproducible simulations. 
 
> The better way is to implement gen_seed(rng_type::Type{T}) where T<:RandomNumbers.AbstractRNG on the types instead of instances.

OK, but also consider adding `gen_seed(rng::RandomNumbers.AbstractRNG) = gen_seed(typeof(rng))`

**Comment by sunoru** [comment]
> OK, but also consider adding gen_seed(rng::RandomNumbers.AbstractRNG) = gen_seed(typeof(rng))

I see, but I don't know if it's OK to adding that because of https://docs.julialang.org/en/stable/manual/style-guide/#Avoid-confusion-about-whether-something-is-an-instance-or-a-type-1
